gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -110,7 +117,13 @@ public ProxyBackendConfiguration updateBackend(ProxyBackendConfiguration backend
             dao.create(backend.getName(), backend.getRoutingGroup(), backendProxyTo, backendExternalUrl, backend.isActive());
         }
         else {
+            Boolean prevStatus = dao.findFirstByName(backend.getName()).active();
```

Comment:
We already have the model here:
```suggestion
            Boolean prevStatus = model.active();
```